### PR TITLE
Extract feature flag code to a separate `Experiments` framework

### PR DESCRIPTION
**Pull request by jaclync**
Ref p91TBi-68e-p2

## Why

To support push notifications for all stores #5032, we are adding a new [notification service app extension](https://developer.apple.com/documentation/usernotifications/modifying_content_in_newly_delivered_notifications). In order to use the feature flag inside an app extension, having the feature flag code in a separate Xcode framework seems like the ideal solution. This PR extracts the minimum amount of feature flag code from the main app to a separate `Experiments` framework, and allows any new features in an app extension or framework to use the feature flag code.

## Changes

- Created a new Xcode framework called Experiments (unfortunately, the new project file has ~600 diffs) with documentation (`EXPERIMENTS.md`)
- Moved 4 files (`FeatureFlag`, `BuildConfiguration`, `FeatureFlagService`, `DefaultFeatureFlagService`) to Experiments framework, and marked APIs as public when needed
- Moved 3 helpers originally in `BuildConfiguration` to a new file `ProcessConfiguration` in the main app since they aren't based on build configurations
- Imported Experiments framework in the main app when needed

## Testing

I will ensure all checks below pass when the PR is approved, feel free to just review the checklist!

- [x] All tests pass locally
- [x] All tests pass on CI
- [x] All optional UI tests pass on CI
- [x] Installable build works on CI, and the feature flags are working in the installable build as expected (e.g. barcode scanner CTA is visible on the products tab)
- [x] Screenshots generation works as before (depends on https://github.com/woocommerce/woocommerce-ios/pull/5123)

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: docs/EXPERIMENTS.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Experiments
```
Comment (shiki) [inline docs/EXPERIMENTS.md:1]: Just driving by to say: **Thank you** for writing documentation! ❤️

**Comment by jaclync** [comment]
Hi @pmusolino! I'm preparing for [another PR](https://github.com/woocommerce/woocommerce-ios/compare/issue/5032-feature-flag-framework...issue/5032-notification-service-extension-skeleton?expand=1) that depend on this PR, because there are a bunch of project level changes. And there's the third PR coming after that 😅 It'd be great if this PR can be reviewed this week, and if you're busy I can ask other iOS devs too!

Most of the diffs in this PR are on the project file, it's inevitable when creating a new framework (unless we use SPM) 😞

**Comment by pmusolino** [comment]
I'm going to review it right now 🙇

**Review by pmusolino — APPROVED** [review]
Great job @jaclync on that! I didn't notice any issue, and I flagged all the checks list. 

:shipit:

**Comment by jaclync** [comment]
Thank you for the review Paolo! I just made sure the optional UI tests and installable build from the latest commit passed the check. Merging now!

**Pull request merged**